File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.cpp
class NumArray {
    vector<int>s;
public:
    NumArray(vector<int>& nums) {
        int total{};
        for(int n:nums){
            total+=n;
            s.push_back(total);
        }
    }
    
    int sumRange(int left, int right) {
        int sright = s[right];
        int sleft = left>0?s[left-1]:0;
        return sright - sleft;
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefix = []
        total = 0
        for n in nums:
            total+=n
            self.prefix.append(total)

    def sumRange(self, left: int, right: int) -> int:
        rprefix = self.prefix[right]
        lprefix = self.prefix[left-1] if left>0 else 0
        return (rprefix-lprefix)


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)